### DISCUSSION TITLE
How to prevent right click to open jupyterlab's menu?

### DISCUSSION THREAD

**Discussion by Roy-Kid** [manzt/anywidget#720] [General]
Hi, I wonder how to prevent right click to open jupyter menu, but just affects the widget itself. I find this snippet from https://github.com/superstar54/weas-widget/blob/fea56fce2c4a7070f1cbd5803694f243548a94fb/js/widget.js#L244, and apply on `el` of `render({model, el})`, but it doesn't work. Can I ask how to fix it?

**Comment by gereleth** [comment]
You probably need to stop propagation of the `contextmenu` event. 
Something like this: `el.addEventListener("contextmenu", e=>e.stopPropagation())`

**Reply by Roy-Kid** [reply]
you're a genius, bro! fixed!
```
function preventEventPropagation(element: HTMLElement) {
  const stopPropagation = (e) => e.stopPropagation();
  ["contextmenu"].forEach((eventType) => {
    element.addEventListener(eventType, stopPropagation, false);
  });
}
```

![image](https://github.com/user-attachments/assets/3ba77eb3-bb0a-4ea9-8125-f5090c2edf0e)

**Reply by Roy-Kid** [reply]
But no matter which part I select in the cell's output, once I press "R," the cell type will be converted to raw, even if I add "keypress" and "key up " to the list. If you happen to know the answer, could you please help me again?

Also, I believe it's a good feature for those developers whose's widget need to use shortcut?

**Reply by gereleth** [reply]
You're probably looking for `keydown` event this time =)
